### PR TITLE
migrate cleanup-dependencies-project to pekko

### PR DESCRIPTION
**Pull request by laglangyue**
## This pr includes:
 migrate cleanup-dependencies-project to pekko
## Motivation:
it is a subtask of migrating akka to pekko. 
Migrating Pekko from Akka at once will result in many changes and cause trouble for reviewers
## Test:
test localy，there still some problems

### CODE REVIEW

File: docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/build.sbt [hunk 1]
```diff
@@ -50,51 +50,51 @@ dockerRepository := sys.props.get("docker.registry")
...
-  +  "com.lightbend.akka" %% "akka-projection-jdbc" % AkkaProjectionVersion,
+  -  "org.apache.pekko" %% "pekko-projection-cassandra" % PekkoProjectionVersion,
+  +  "org.apache.pekko" %% "pekko-projection-jdbc" % PekkoProjectionVersion,
   // end::replace-offset-store-for-projections-jdbc[]
    */
-  "com.typesafe.akka" %% "akka-http" % AkkaHttpVersion,
+  "org.apache.pekko" %% "pekko-http" % PekkoHttpVersion,
   // tag::remove-grpc-optional[]
-  "com.typesafe.akka" %% "akka-http2-support" % AkkaHttpVersion,
+  // "org.apache.pekko" %% "pekko-http2-support" % PekkoHttpVersion,
```
Comment (laglangyue) [inline docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/build.sbt:79]: well I didn't find pekko-http2-support，Is this included in pekko http or not published ?
@pjfanning

**Comment by laglangyue** [comment]
I didn;t find a `sbt-pekko-grpc`
```
// tag::remove-grpc-plugin[]
addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "2.1.5")
// end::remove-grpc-plugin[]
```

File: docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/build.sbt (hunk 1 above)
Comment (pjfanning) [reply docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/build.sbt:79]: not published - not needed for pekko (code was moved to pekko-http-core)

**Comment by pjfanning** [comment]
> I didn't find a `sbt-pekko-grpc`
> 
> ```
> // tag::remove-grpc-plugin[]
> addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "2.1.5")
> // end::remove-grpc-plugin[]
> ```

Use
```
addSbtPlugin("org.apache.pekko" % "pekko-grpc-sbt-plugin" % "1.0.2")
```

File: docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/project/plugins.sbt [hunk 2]
```diff
@@ -1,5 +1,5 @@
 // tag::remove-grpc-plugin[]
-addSbtPlugin("com.lightbend.akka.grpc" % "sbt-akka-grpc" % "2.1.5")
+addSbtPlugin("org.apache.pekko" % "pekko-grpc-sbt-plugin" % "1.0.0")
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/project/plugins.sbt:2]: 1.0.1 please

Comment (pjfanning) [reply docs-source/docs/modules/how-to/examples/cleanup-dependencies-project/project/plugins.sbt]: I changed this myself

**Comment by laglangyue** [comment]
1.0.1 of pekko-grpc-sbt-plugin will cause conflict
and 1.0.0 is ok
```shell
[error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
[error] 
[error] 	* org.scala-lang.modules:scala-xml_2.12:2.2.0 (early-semver) is selected over 1.1.1
[error] 	    +- com.typesafe.play:twirl-api_2.12:1.6.1             (depends on 2.2.0)
[error] 	    +- com.typesafe.sbt:sbt-native-packager:1.8.1 (sbtVersion=1.0, scalaVersion=2.12) (depends on 1.1.1)
```

**Comment by pjfanning** [comment]
> 1.0.1 of pekko-grpc-sbt-plugin will cause conflict and 1.0.0 is ok
> 
> ```shell
> [error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
> [error] 
> [error] 	* org.scala-lang.modules:scala-xml_2.12:2.2.0 (early-semver) is selected over 1.1.1
> [error] 	    +- com.typesafe.play:twirl-api_2.12:1.6.1             (depends on 2.2.0)
> [error] 	    +- com.typesafe.sbt:sbt-native-packager:1.8.1 (sbtVersion=1.0, scalaVersion=2.12) (depends on 1.1.1)
> ```

I added https://github.com/apache/incubator-pekko-platform-guide/pull/57/commits/23c29bf4c28906ab9811b2eb8aeadc5e430289a7 - we've had to do that in other builds too

**Comment by laglangyue** [comment]
> > 1.0.1 of pekko-grpc-sbt-plugin will cause conflict and 1.0.0 is ok
> > ```shell
> > [error] (update) found version conflict(s) in library dependencies; some are suspected to be binary incompatible:
> > [error] 
> > [error] 	* org.scala-lang.modules:scala-xml_2.12:2.2.0 (early-semver) is selected over 1.1.1
> > [error] 	    +- com.typesafe.play:twirl-api_2.12:1.6.1             (depends on 2.2.0)
> > [error] 	    +- com.typesafe.sbt:sbt-native-packager:1.8.1 (sbtVersion=1.0, scalaVersion=2.12) (depends on 1.1.1)
> > ```
> 
> I added [23c29bf](https://github.com/apache/incubator-pekko-platform-guide/commit/23c29bf4c28906ab9811b2eb8aeadc5e430289a7) - we've had to do that in other builds too

should we use this,sbt-native-packager maybe remove to it,but not `com.typesafe.sbt`
addSbtPlugin("com.github.sbt" % "sbt-native-packager" % "1.9.16")

**Comment by laglangyue** [comment]
yeah, sbt-native-packager was migrated to 
// https://mvnrepository.com/artifact/com.github.sbt/sbt-native-packager_2.12_1.0
libraryDependencies += "com.github.sbt" % "sbt-native-packager_2.12_1.0" % "1.9.16"
this is rep
https://github.com/sbt/sbt-native-packager

**Comment by laglangyue** [comment]
Please trigger the workflow
@pjfanning 
PTAL, waitting for CI.
I would away from my compute and go home.If there are still any problems, I will handle them tomorrow.

**Pull request merged**